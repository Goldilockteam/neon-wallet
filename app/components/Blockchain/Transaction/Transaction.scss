@import '../../../styles/variables';

.txidLink {
  color: $link-color;
  font-size: 13px;
  cursor: pointer;
}

.txDateContainer {
  width: 150px;
  text-align: center;
  font-size: 13px;
  font-family: Gotham-Book;
}

.txLabelContainer {
  font-family: Gotham-Book;
  font-size: 14px;
  text-align: center;
  width: 75px;
}

.txAmountContainer {
  width: 115px;
  text-align: center;
  font-weight: 400;
}

.txToContainer {
  width: 285px;
  text-align: center;
  font-size: 12px;
}

.transactionContainer {
  display: flex;
  align-items: center;
  padding: 0 20px;
  justify-content: space-evenly;
  width: 100%;
}

.invocationTransaction {
  font-size: 13px;
}

.amounts {
  align-items: center;
  display: flex;
  width: 255px;
  justify-content: center;
}

.txTypeIconContainer {
  width: 50px;
  .claimIconContainer {
    svg {
      path {
        fill: #ffcf48;
      }
    }
  }

  .sendIconContainer {
    svg {
      path {
        fill: #ff967e;
      }
    }
  }

  .receiveIconContainer {
    svg {
      path {
        fill: #82e092;
      }
    }
  }
}
