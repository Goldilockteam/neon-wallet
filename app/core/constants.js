// @flow
import tokenList from './tokenList.json'
import nodesMainNet from './nodes-main-net.json'
import nodesTestNet from './nodes-test-net.json'

export const NEON_WALLET_RELEASE_LINK =
  'https://github.com/CityOfZion/neon-wallet/releases'

export const DISCORD_INVITE_LINK = 'https://discordapp.com/invite/R8v48YA'
export const COZ_DONATIONS_ADDRESS = 'Adr3XjZ5QDzVJrWvzmsTTchpLRRGSzgS5A'

export const ASSETS = {
  NEO: 'NEO',
  GAS: 'GAS'
}

export const THEMES = {
  LIGHT: 'Light',
  DARK: 'Dark'
}

export const DEFAULT_THEME = THEMES.LIGHT

export const EXPLORERS = {
  NEO_SCAN: 'Neoscan',
  NEO_TRACKER: 'Neotracker',
  ANT_CHAIN: 'Antchain'
}

export const DEFAULT_EXPLORER = EXPLORERS.NEO_SCAN

export const ROUTES = {
  HOME: '/',
  DASHBOARD: '/dashboard',
  RECEIVE: '/receive',
  CONTACTS: '/contacts',
  ADD_CONTACT: '/contacts/new',
  EDIT_CONTACT: '/contacts/edit/:name',
  CREATE_WALLET: '/create',
  CREATE_WALLET_AUTHENTICATED: '/create-authenticated',
  IMPORT_WALLET: '/import',
  IMPORT_WALLET_AUTHENTICATED: '/import-authenticated',
  TOKEN_SALE: '/token-sale',
  TOKEN_SALE_CONFIRMATION: '/token-sale-confirm',
  TOKEN_SALE_FAILURE: '/token-sale-failure',
  TOKEN_SALE_SUCCESS: '/token-sale-success',
  TRANSACTION_HISTORY: '/transactions',
  SETTINGS: '/settings',
  DISPLAY_WALLET_KEYS: '/display-wallet-keys',
  DISPLAY_WALLET_KEYS_AUTHENTICATED: '/display-wallet-keys-authenticated',
  DISPLAY_WALLET_QRS: '/display-wallet-qrs',
  DISPLAY_WALLET_QRS_AUTHENTICATED: '/display-wallet-qrs-authenticated',
  WALLET_MANAGER: '/wallet-manager',
  EDIT_WALLET: '/edit-wallet/:key/:label',
  SEND: '/send/',
  SEND_ADDRESS: '/send/:address',
  ENCRYPT: '/encrypt-private-key',
  NODE_SELECT: '/node-select',
  VOTING: '/voting'
}

export const NOTIFICATION_LEVELS = {
  ERROR: 'error',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning'
}

export const NOTIFICATION_POSITIONS = {
  TOP_CENTER: 'tc',
  TOP_RIGHT: 'tr',
  TOP_LEFT: 'tl',
  BOTTOM_CENTER: 'bc',
  BOTTOM_RIGHT: 'br',
  BOTTOM_LEFT: 'bl'
}

export const BIP44_PATH = '8000002C' + '80000378' + '80000000' + '00000000' // eslint-disable-line

export const MODAL_TYPES = {
  SEND: 'SEND',
  CONFIRM: 'CONFIRM',
  TOKEN_INFO: 'TOKEN_INFO',
  TOKEN: 'TOKEN',
  ICO: 'ICO',
  ADD_CONTACT: 'ADD_CONTACT',
  RECEIVE: 'RECEIVE'
}

// export const MAIN_NETWORK_ID = '1'
export const TEST_NETWORK_ID = '2'
// export const COZ_TEST_NETWORK_ID = '3'
// export const GOLDI_NETWORK_ID = '4'

// TestNet
export const TOKENS_TEST = {
  DBC: 'b951ecbbc5fe37a9c280a76cb0ce0014827294cf',
  RPX: '5b7074e873973a6ed3708862f219a6fbf4d1c411',
  QLC: '0d821bd7b6d53f5c2b40e217c6defc8bbe896cf5',
  LOCK: 'c36a622ad044df617634460a0f68e9023c0aa4aa'
}

// MainNet
export const TOKENS = tokenList

// GoldiNet
export const TOKENS_GOLDI = {
}

export const ENDED_ICO_TOKENS = [
  'DBC',
  'RPX',
  'RHT',
  'QLC',
  'NRVE',
  'IAM',
  'ONT',
  'THOR',
  'CGE',
  'AVA',
  'SWH',
  'SWTH',
  'EFX',
  'MCT',
  'PKC'
]

export const DEFAULT_WALLET = {
  name: 'userWallet',
  version: '1.0',
  scrypt: {
    cost: 16384,
    blockSize: 8,
    parallel: 8,
    size: 64
  },
  accounts: [],
  extra: null
}

export const DEFAULT_CURRENCY_CODE = 'usd'

export const CURRENCIES = {
  aud: { symbol: '$' },
  brl: { symbol: 'R$' },
  cad: { symbol: '$' },
  chf: { symbol: 'Fr.' },
  clp: { symbol: '$' },
  cny: { symbol: '¥' },
  czk: { symbol: 'Kč' },
  dkk: { symbol: 'kr. ' },
  eur: { symbol: '€' },
  gbp: { symbol: '£' },
  hkd: { symbol: '$' },
  huf: { symbol: 'Ft ' },
  idr: { symbol: 'Rp ' },
  ils: { symbol: '₪' },
  inr: { symbol: '₹' },
  jpy: { symbol: '¥' },
  krw: { symbol: '₩' },
  mxn: { symbol: '$' },
  myr: { symbol: 'RM' },
  nok: { symbol: 'kr ' },
  nzd: { symbol: '$' },
  php: { symbol: '₱' },
  pkr: { symbol: '₨ ' },
  pln: { symbol: 'zł' },
  rub: { symbol: '₽' },
  sek: { symbol: 'kr ' },
  sgd: { symbol: 'S$' },
  thb: { symbol: '฿' },
  try: { symbol: '₺' },
  twd: { symbol: 'NT$' },
  usd: { symbol: '$' },
  zar: { symbol: 'R ' }
}

export const TOKEN_SALE_PURCHASE = 'token_sale_purchase'
export const TOKEN_SALE_CONFIRM = 'token_sale_confirm'
export const TOKEN_SALE_SUCCESS = 'token_sale_success'
export const TOKEN_SALE_FAILURE = 'token_sale_failure'

export const NODES_MAIN_NET = nodesMainNet
export const NODES_TEST_NET = nodesTestNet

export const PRICE_UNAVAILABLE = 'N/A'
