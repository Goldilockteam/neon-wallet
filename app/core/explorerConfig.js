import { EXPLORERS } from './constants'
const { ANT_CHAIN, NEO_SCAN, NEO_TRACKER } = EXPLORERS

export default {
  [ANT_CHAIN]: {
    addressLinkStructure: 'address/info/',
    assetLinkStructure: 'asset/hash/',
    mainNetwork: 'http://antcha.in/',
    testNetwork: 'http://testnet.antcha.in/',
    trxLinkStructure: 'tx/hash/0x'
  },
  [NEO_SCAN]: {
    addressLinkStructure: 'address/',
    assetLinkStructure: 'asset/',
    mainNetwork: 'https://neoscan.io/',
    testNetwork: 'https://neoscan-testnet.io/',
    trxLinkStructure: 'transaction/'
  },
  [NEO_TRACKER]: {
    addressLinkStructure: 'address/',
    assetLinkStructure: 'asset/',
    mainNetwork: 'https://neotracker.io/',
    testNetwork: 'https://testnet.neotracker.io/',
    trxLinkStructure: 'tx/'
  }

}
