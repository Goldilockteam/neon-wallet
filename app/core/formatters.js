// @flow
import { ASSETS } from './constants'
import { toBigNumber } from './math'

export const COIN_DECIMAL_LENGTH = 8
export const SHORT_DISPLAY_DECIMAL_LENGTH = 4

type ValueType = string | number

export const formatGAS = (
  value: ValueType,
  shortDisplay: boolean = false
): string => {
  const decimals = shortDisplay
    ? SHORT_DISPLAY_DECIMAL_LENGTH
    : COIN_DECIMAL_LENGTH
  return toBigNumber(value).toFormat(decimals)
}

export const formatToRoundedShortNumber = (value: number): string => {
  if (Number.isInteger(value)) return value.toString()
  return toBigNumber(value)
    .toFormat(SHORT_DISPLAY_DECIMAL_LENGTH)
    .toString()
}

export const formatThousands = (value: ValueType): string =>
  toBigNumber(value).toFormat(0)

export const formatNEO = (value: ValueType): string =>
  toBigNumber(value).toFormat(0)

export const formatBalance = (
  symbol: SymbolType,
  balance: ValueType,
  shortDisplay: boolean = false
): string => {
  if (symbol === ASSETS.NEO) {
    return formatNEO(balance)
  }
  return formatGAS(balance, shortDisplay)
}

export const toFixedDecimals = (
  value: ValueType,
  decimals: number = COIN_DECIMAL_LENGTH
): string => toBigNumber(value).toFixed(decimals)

export const formatFiat = (value: ValueType = 0): string =>
  toBigNumber(value).toFormat(2)
