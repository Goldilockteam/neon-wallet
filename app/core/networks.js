// @flow
import { MAIN_NETWORK_ID, TEST_NETWORK_ID, COZ_TEST_NETWORK_ID, GOLDI_NETWORK_ID } from './constants'

export const isMainNetwork = (networkId: string) =>
  networkId === MAIN_NETWORK_ID
export const isTestNetwork = (networkId: string) =>
  networkId === TEST_NETWORK_ID
export const isCozNetwork = (networkId: string) =>
  networkId === COZ_TEST_NETWORK_ID
export const isGoldiNetwork = (networkId: string) =>
  networkId === GOLDI_NETWORK_ID

export const getNetworks = () => [
  {
    id: MAIN_NETWORK_ID,
    label: 'MainNet',
    network: 'MainNet'
  },
  {
    id: TEST_NETWORK_ID,
    label: 'TestNet',
    network: 'TestNet'
  },
  {
    id: COZ_TEST_NETWORK_ID,
    label: 'CoZ TestNet',
    network: 'CozNet'
  },
  {
    id: GOLDI_NETWORK_ID,
    label: 'GoldiNet',
    network: 'GoldiNet'
  }
]

export const findNetwork = (networkId: string): NetworkItemType => {
  const networks = getNetworks()
  return networks.find(({ id }) => id === networkId) || networks[0]
}
