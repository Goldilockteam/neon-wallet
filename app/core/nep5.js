// @flow
import { map, isEmpty } from 'lodash'
import axios from 'axios'

import { toBigNumber } from './math'
import { COIN_DECIMAL_LENGTH } from './formatters'
import {
  TOKENS,
  TOKENS_TEST,
  TOKENS_GOLDI,
  // MAIN_NETWORK_ID,
  TEST_NETWORK_ID//,
  // GOLDI_NETWORK_ID
} from './constants'

let fetchedTokens

export const adjustDecimalAmountForTokenTransfer = (value: string): string =>
  toBigNumber(value)
    .times(10 ** COIN_DECIMAL_LENGTH)
    .round()
    .toNumber()

const getTokenEntry = ((): Function => {
  let id = 1
  return (
    symbol: string,
    scriptHash: string,
    networkId: string,
    image: string,
    name: string
  ) => ({
    id: `${id++}`, // eslint-disable-line no-plusplus
    symbol,
    scriptHash,
    networkId,
    isUserGenerated: false,
    image,
    name
  })
})()

export const getDefaultTokens = async (): Promise<Array<TokenItemType>> => {
  const tokens = []
  /*
  // Prevent duplicate requests here
  if (!fetchedTokens) {
    const response = await axios
      // use a time stamp query param to prevent caching
      .get(
        `https://raw.githubusercontent.com/CityOfZion/neo-tokens/master/tokenList.json?timestamp=${new Date().getTime()}`
      )
      .catch(error => {
        console.error('Falling back to hardcoded list of NEP5 tokens!', error)
        // if request to gh fails use hardcoded list
        fetchedTokens = TOKENS
      })
    if (response && response.data && !isEmpty(response.data)) {
      fetchedTokens = response.data
    } else {
      fetchedTokens = TOKENS
    }
  }

  tokens.push(
    ...map(fetchedTokens, tokenData =>
      getTokenEntry(
        tokenData.symbol,
        tokenData.networks['1'].hash,
        MAIN_NETWORK_ID,
        tokenData.image,
        tokenData.networks['1'].name
      )
    )
  )
  */
  tokens.push(
    ...map(TOKENS_TEST, (scriptHash, symbol) =>
      getTokenEntry(symbol, scriptHash, TEST_NETWORK_ID)
    )
  )
  /*
  tokens.push(
    ...map(TOKENS_GOLDI, (scriptHash, symbol) =>
      getTokenEntry(symbol, scriptHash, GOLDI_NETWORK_ID)
    )
  )
  */

  return tokens
}
