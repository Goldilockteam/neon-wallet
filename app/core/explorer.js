// @flow
import { openExternal } from './electron'
import explorerConfig from './explorerConfig'
import { isMainNetwork, isGoldiNetwork } from './networks'

export const buildExplorerLink = (
  networkId: string,
  explorer: ExplorerType,
  structure: string,
  suffix: string
) => {
  if (explorerConfig.hasOwnProperty(explorer)) {
    const explorerInfo = explorerConfig[explorer]

    return getExplorerBaseURL(networkId, explorer) + explorerInfo[structure] + suffix
  }

  throw new Error(`${explorer} is not a valid explorer type. Valid explorer types are: ${Object.keys(explorerConfig).join(', ')}`)
}

export const getExplorerBaseURL = (networkId: string, explorer: ExplorerType) => {
  if (explorerConfig.hasOwnProperty(explorer)) {
    const selectedExplorer = explorerConfig[explorer]

    // temporary until neon-wallet implements proper
    // per-network explorer types
    if (isGoldiNetwork(networkId))
      return 'https://neo.goldilock.com'

    return isMainNetwork(networkId)
      ? selectedExplorer.mainNetwork
      : selectedExplorer.testNetwork
  }
}

export const getExplorerTxLink = (networkId: string, explorer: ExplorerType, txId: string) => buildExplorerLink(networkId, explorer, 'trxLinkStructure', txId)

export const getExplorerAddressLink = (networkId: string, explorer: ExplorerType, address: string) => buildExplorerLink(networkId, explorer, 'addressStructure', address)

export const getExplorerAssetLink = (networkId: string, explorer: ExplorerType, assetId: string) => buildExplorerLink(networkId, explorer, 'assetLinkStructure', assetId)

export const openExplorerTx = (networkId: string, explorer: ExplorerType, txId: string) => openExternal(getExplorerTxLink(networkId, explorer, txId))

export const openExplorerAddress = (networkId: string, explorer: ExplorerType, address: string) => openExternal(getExplorerAddressLink(networkId, explorer, address))

export const openExplorerAsset = (networkId: string, explorer: ExplorerType, assetId: string) => openExternal(getExplorerAssetLink(networkId, explorer, assetId))
