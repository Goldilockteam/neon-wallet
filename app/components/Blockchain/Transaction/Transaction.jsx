// @flow
import React, { Fragment } from 'react'
import type { Node } from 'react'

import moment from 'moment'
import { isEmpty } from 'lodash-es'

import Button from '../../Button'
import { openExplorerTx } from '../../../core/explorer'
import styles from './Transaction.scss'
import ClaimIcon from '../../../assets/icons/claim.svg'
import SendIcon from '../../../assets/icons/send-tx.svg'
import ReceiveIcon from '../../../assets/icons/receive-tx.svg'
import ContactsAdd from '../../../assets/icons/contacts-add.svg'
import InfoIcon from '../../../assets/icons/info.svg'
import CopyToClipboard from '../../CopyToClipboard'
import Tooltip from '../../Tooltip'

type Props = {
  tx: Object,
  networkId: string,
  explorer: ExplorerType,
  contacts: Object,
  showAddContactModal: ({ address: string }) => null
}

export default class Transaction extends React.Component<Props> {
  render = () => {
    const { tx } = this.props
    const { iconType } = tx
    return (
      <div className={styles.transactionContainer}>
        {this.renderAbstract(iconType)}
        <Button
          className={styles.transactionHistoryButton}
          renderIcon={InfoIcon}
          onClick={this.handleClick}
        >
          View
        </Button>
      </div>
    )
  }

  findContact = (address: string): Node => {
    const { contacts } = this.props
    if (contacts && !isEmpty(contacts)) {
      let label
      Object.keys(contacts).forEach(key => {
        if (contacts[key] === address) {
          label = key
        }
      })
      return label ? (
        <Tooltip title={address} className={styles.largerFont}>
          {label}
        </Tooltip>
      ) : (
        address
      )
    }
    return address
  }

  displayModal = () => {
    const {
      showAddContactModal,
      tx: { to, from, iconType }
    } = this.props
    showAddContactModal({ address: iconType === 'RECEIVE' ? from : to })
  }

  handleClick = () => {
    const { networkId, explorer, tx } = this.props
    const { txid } = tx
    openExplorerTx(networkId, explorer, txid)
  }

  renderAbstract = (type: string) => {
    const { tx } = this.props
    const { iconType, time, label, amount, isNetworkFee, to, from } = tx
    switch (type) {
      case 'CLAIM':
        return (
          <div className={styles.abstractContainer}>
            <div className={styles.txTypeIconContainer}>
              <div className={styles.claimIconContainer}>
                <ClaimIcon />
              </div>
            </div>
            <div className={styles.txDateContainer}>
              {moment.unix(time).format('MM/DD/YYYY | HH:MM:ss')}
            </div>
            <div className={styles.txLabelContainer}>{label}</div>
            <div className={styles.txAmountContainer}>{amount}</div>
            <div className={styles.txToContainer}>
              <Fragment>
                {this.findContact(to)}
                {this.findContact(to) === to && (
                  <CopyToClipboard
                    className={styles.copy}
                    text={to}
                    tooltip="Copy Public Address"
                  />
                )}
              </Fragment>
            </div>
            <div className={styles.historyButtonPlaceholder} />
          </div>
        )
      case 'SEND':
        return (
          <div className={styles.abstractContainer}>
            <div className={styles.txTypeIconContainer}>
              <div className={styles.sendIconContainer}>
                <SendIcon />
              </div>
            </div>
            <div className={styles.txDateContainer}>
              {moment.unix(time).format('MM/DD/YYYY | HH:MM:ss')}
            </div>
            <div className={styles.txLabelContainer}>{label}</div>
            <div className={styles.txAmountContainer}>{amount}</div>
            <div className={styles.txToContainer}>
              {isNetworkFee ? (
                <div className={styles.largerFont}> {to} </div>
              ) : (
                <Fragment>
                  {this.findContact(to)}
                  {this.findContact(to) === to && (
                    <CopyToClipboard
                      className={styles.copy}
                      text={to}
                      tooltip="Copy Public Address"
                    />
                  )}
                </Fragment>
              )}
            </div>
            {isNetworkFee ? (
              <div className={styles.historyButtonPlaceholder} />
            ) : (
              <Button
                className={styles.transactionHistoryButton}
                renderIcon={ContactsAdd}
                onClick={this.displayModal}
                disabled={this.findContact(to) !== to}
              >
                Add
              </Button>
            )}
          </div>
        )
      case 'RECEIVE':
        return (
          <div className={styles.abstractContainer}>
            <div className={styles.txTypeIconContainer}>
              <div className={styles.receiveIconContainer}>
                <ReceiveIcon />
              </div>
            </div>
            <div className={styles.txDateContainer}>
              {moment.unix(time).format('MM/DD/YYYY | HH:MM:ss')}
            </div>
            <div className={styles.txLabelContainer}>{label}</div>
            <div className={styles.txAmountContainer}>{amount}</div>
            <div className={styles.txToContainer}>
              {this.findContact(from)}
              {this.findContact(from) === from &&
                from !== 'MINT TOKENS' && (
                  <CopyToClipboard
                    className={styles.copy}
                    text={from}
                    tooltip="Copy Public Address"
                  />
                )}
            </div>
            {this.findContact(from) !== from ? (
              <div className={styles.transactionHistoryButton} />
            ) : (
              <Button
                className={styles.transactionHistoryButton}
                renderIcon={ContactsAdd}
                onClick={this.displayModal}
              >
                Add
              </Button>
            )}
          </div>
        )

      default:
        console.warn('renderTxTypeIcon() invoked with an invalid argument!', {
          type
        })
        return null
    }
  }
}
