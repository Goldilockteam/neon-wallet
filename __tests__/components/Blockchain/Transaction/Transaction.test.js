import React from 'react'
import { shallow } from 'enzyme'

import Transaction from '../../../../app/components/Blockchain/Transaction/Transaction'
import { TEST_NETWORK_ID, EXPLORERS } from '../../../../app/core/constants'

describe('Transaction', () => {
  const props = {
    explorer: EXPLORERS.NEO_SCAN,
    networkId: TEST_NETWORK_ID,
    txid: 'b1b4be0852c402d6d2609c2026b700702333e32a4bbf4f9e13ebe07699323951'
  }

  test('should render without crashing', () => {
    const wrapper = shallow(<Transaction { ...props } />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should handle the click event', () => {
    const spy = jest.spyOn(Transaction.prototype, 'handleClick')
    const wrapper = shallow(<Transaction { ...props } />)
    wrapper.simulate('click')
    expect(spy).toHaveBeenCalled();
  });
});
