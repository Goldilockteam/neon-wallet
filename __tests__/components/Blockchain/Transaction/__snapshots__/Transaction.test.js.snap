// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Transaction should render without crashing 1`] = `
<span
  className="transaction"
  onClick={[Function]}
>
  b1b4be0852c402d6d2609c2026b70070
</span>
`;
